src/smc-webapp/buttonbar.coffee

```diff
@@ -51,11 +51,280 @@ exports.commands =
                 right : ''
         set_name_and_email :
                 insert:
-                         """
-                         git config --global user.name ""
-                         git config --global user.email ""
-                         """
+                        """
+                        git config --global user.name ""
+                        git config --global user.email ""
+                        """
+        initalize_git :
+                insert: 
+                        """
+                        git init
+                        """  
+        create_gitignore :
+                insert:
```

Comment:
And maybe add a comment with a link to a website full of gitignore files...?

On Sat, May 14, 2016 at 9:45 AM, Harald Schilly <notifications@github.com>
wrote:

> In src/smc-webapp/buttonbar.coffee
> <https://github.com/sagemathinc/smc/pull/551#discussion_r63278744>:
>
> > @@ -51,11 +51,280 @@ exports.commands =
> >                  right : ''
> >          set_name_and_email :
> >                  insert:
> > -                         """
> > -                         git config --global user.name ""
> > -                         git config --global user.email ""
> > -                         """
> > +                        """
> > +                        git config --global user.name ""
> > +                        git config --global user.email ""
> > +                        """
> > +        initalize_git :
> > +                insert:
> > +                        """
> > +                        git init
> > +                        """
> > +        create_gitignore :
> > +                insert:
>
> for me, this gitignore is way too large. please reduce it to the minimals,
> like *.py[cod] etc.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/sagemathinc/smc/pull/551/files/4bf0e9292a8d78b83055d528129ad7e2a76560a3#r63278744>
>



-- 
William (http://wstein.org)